src/pages/Signin/SigninPage.jsx

```diff
@@ -1,138 +1,165 @@
-import React, { useState } from 'react'
+import React, { useState } from "react";
+import { useNavigate } from "react-router-dom";
 import {
-    InputAdornment,
-    IconButton,
-    OutlinedInput,
-    InputLabel,
-    Typography,
-    Divider,
-    Link,
-    Grid,
-    Alert,
-    TextField,
-    Button
-} from '@mui/material'
-import { Visibility, VisibilityOff } from '@mui/icons-material';
-import { signInWithEmailAndPassword, signInWithPopup } from 'firebase/auth';
-import { auth, googleProvider } from '../../../firebase.config'
-import { GridStyled, LogoStyled, OutlinedInputStyled, TitleStyled, TopicImageStyled } from './styles';
-
+  InputAdornment,
+  IconButton,
+  OutlinedInput,
+  InputLabel,
+  Typography,
+  Divider,
+  Link,
+  Grid,
+  Alert,
+  TextField,
+  Button,
+  useMediaQuery
+} from "@mui/material";
+import { Visibility, VisibilityOff } from "@mui/icons-material";
+import { signInWithEmailAndPassword, signInWithPopup } from "firebase/auth";
+import { auth, googleProvider } from "../../../firebase.config";
+import {
+  TopicImageGrid,
+  GridStyled,
+  LogoStyled,
+  OutlinedInputStyled,
+  TitleStyled,
+  TopicImageStyled,
+  InputGrid,
+} from "./styles";
 
 export default function SigninPage() {
-    // Hooks
-    const [email, setEmail] = useState("");
-    const [isAuth, setIsAuth] = useState(true);
-    const [password, setPassword] = useState("");
-    const [showPassword, setShowPassword] = useState(false);
+  // Hooks
+  const [email, setEmail] = useState("");
```

Comment:
You can also use a single state like

const [userData, setUserData] = useState({ })